src/mongoc/mongoc-server-description.c

```diff
@@ -179,13 +189,33 @@ mongoc_server_description_has_rs_member(mongoc_server_description_t *server,
          bson_iter_init (&member_iter, rs_members[i]);
 
          while (bson_iter_next (&member_iter)) {
-            if (strcmp (address, bson_iter_utf8 (&member_iter, NULL)) == 0) {
+            /* TODO we convert to lowercase each time we do this, it might be
+               better to store lowercased member names instead */
+            name = bson_iter_utf8 (&member_iter, NULL);
+            name_len = strlen (name);
+
+            /* re-use previously malloced buffer if we can */
+            if (name_len > buf_len) {
+               if (buf) bson_free (buf);
```

Comment:
You can just malloc two buffers the size of address.  If the member you're looking at is larger than address (that second parameter to bson_iter_utf8()), you don't need to lower case it to need it's different.  

More generally, you only need to strcmp strings that are the same size.  Otherwise they're obviously different if they're different lengths